element_calcium_imaging/imaging_report.py

```diff
@@ -0,0 +1,414 @@
+            "zoomOut",
+            "autoScale2d",
+        ],
+    )
+
+
+@schema
+class ScanLevelReport(dj.Computed):
+    definition = """
+    -> imaging.Segmentation
+    ---
+    average_image: longblob
+    """
+
+    def make(self, key):
+        image = (imaging.MotionCorrection.Summary & key).fetch1("average_image")
+
+        cell_mask_ids, mask_xpix, mask_ypix = (
+            imaging.Segmentation.Mask * imaging.MaskClassification.MaskType & key
+        ).fetch("mask", "mask_xpix", "mask_ypix")
+
+        background_image = single_to_3channel_image(image, low_q=0, high_q=99.9)
+
+        background_image_with_cells_painted = paint_rois(
+            background_image, mask_xpix, mask_ypix
+        )
+        cells_maskid_image = make_maskid_image(
+            background_image[:, :, 0], cell_mask_ids, mask_xpix, mask_ypix
+        )
+        background_with_cells = alpha_combine_2images(
+            background_image, background_image_with_cells_painted
+        )
+
+        image_fig = go.Figure(
+            go.Image(
+                z=background_with_cells,
+                hovertemplate="x: %{x} <br>y: %{y} <br>mask_id: %{customdata}",
+                customdata=cells_maskid_image,
+            )
+        )
+
+        image_fig.update_layout(
+            title="Average Image with Cells",
+            xaxis={
+                "title": "X (px)",
+                "visible": True,
+                "showticklabels": True,
+                "showgrid": False,
+            },
+            yaxis={
+                "title": "Y (px)",
+                "visible": True,
+                "showticklabels": True,
+                "showgrid": False,
+            },
+            paper_bgcolor="rgba(0,0,0,0)",
+            plot_bgcolor="rgba(0,0,0,0)",
+        )
+
+        self.insert1({**key, "average_image": image_fig.to_json()})
+
+
+@schema
+class ActivityReport(dj.Computed):
+    definition = """
+    -> imaging.Segmentation.Mask
+    ---
+    activity_trace: longblob
+    """
+
+    def make(self, key):
+        activity_trace = (
+            imaging.Activity.Trace & "extraction_method LIKE '%deconvolution'" & key
+        ).fetch1("activity_trace")
+        fluorescence, fps = (scan.ScanInfo * imaging.Fluorescence.Trace & key).fetch1(
+            "fluorescence", "fps"
+        )
+
+        trace_fig = go.Figure(
+            [
+                go.Scatter(
+                    x=np.arange(len(fluorescence)) / fps,
+                    y=fluorescence,
+                    name="Fluorescence",
+                    yaxis="y1",
+                ),
+                go.Scatter(
+                    x=np.arange(len(activity_trace)) / fps,
+                    y=activity_trace,
+                    name="Calcium Event",
+                    yaxis="y2",
+                ),
+            ]
+        )
+
+        trace_fig.update_layout(get_tracelayout(key))
+
+        self.insert1({**key, "activity_trace": trace_fig.to_json()})
```

Comment:
@tdincer and I have discussed this, but reiterating here for reference.
Storing the JSON formatted string in DB as longblob is more performant (on insert and fetch) compared to that of storing as python dictionary (`json.loads(trace_fig.to_json())`). However, currently sciviz plotly rendering does not support fetching and rendering with JSON string, pharus expects a python dictionary to be returned upon fetch (and then does the conversion to JSON string).

This behavior can be improved quite easily, perhaps adding an `isinstance()` check here: https://github.com/datajoint/pharus/blob/ccef4ecbc163dcb1f80ea0ad103da67e8e5bbff1/pharus/component_interface.py#L375